src/encoded/types/file.py

```diff
@@ -313,7 +313,123 @@ class MatrixFile(AnalysisFile):
     rev = {
         'cell_annotations': ('CellAnnotation', 'matrix_files'),
         'quality_metrics': ('Metrics', 'quality_metric_of')
-    
+    }
+
+
+    @calculated_property(schema={
+        "title": "Quality metrics",
+        "description": "The list of QC metric objects associated with this file.",
+        "comment": "Do not submit. Values in the list are reverse links of a quality metric with this file in quality_metric_of field.",
+        "type": "array",
+        "items": {
+            "type": ['string', 'object'],
+            "linkFrom": "Metrics.quality_metric_of",
+        },
+        "notSubmittable": True,
+    })
+    def quality_metrics(self, request, quality_metrics=None):
+        if quality_metrics:
+            return paths_filtered_by_status(request, quality_metrics)
+
+
+    @calculated_property(schema={
+        "title": "Assays",
+        "description": "The list of assays used to generate data contained in this matrix.",
+        "comment": "Do not submit. Values in the list are reverse links of a quality metric with this file in quality_metric_of field.",
```

Comment:
_Values in the list are reverse links of a quality metric with this file in quality_metric_of field._
I am guessing it's the "reverse calculated fields of assay from library objects"?

Reply:
Good catch. Updated x3 to "Do not submit. Values in the list are inherited from linked Library objects."